addons/mail/static/tests/mock_server/mail_mock_server.js

```diff
@@ -724,12 +698,18 @@ async function mail_thread_messages(request) {
 registerRoute("/mail/action", mail_action);
 /** @type {RouteCallback} */
 async function mail_action(request) {
+    if (dataDeferred) {
+        await dataDeferred;
+    }
```

Comment:
Those are very generic routes, it's not very desirable to arbitrarily block the next call, it should be more related to a specific data request.